src/batch_request.cpp

```diff
@@ -17,14 +17,14 @@
 #include "batch_request.hpp"
 
 #include "execute_request.hpp"
+#include "external_types.hpp"
 #include "serialization.hpp"
 #include "statement.hpp"
-#include "types.hpp"
 
 
 extern "C" {
 
-CassBatch* cass_batch_new(CassBatchType type) {
+CassBatch* cass_batch_new(CassSession* session, CassBatchType type) {
```

Comment:
That sounds like a reasonable approach. Just some other thoughts for brainstorming:

The slow path could become more common as the cpp driver catches up to bleeding edge...

Could there be a global default setting controlling the max protocol for default encoding (removing the penalty for clients that configure it to their version).

Another option would be to defer encoding until the message is going to be written. May be less attractive since this might require copying values in, though.